zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/controller/TopologyControllerTest.java

```diff
@@ -5,115 +5,131 @@
  * Licensed under the Camunda License 1.0. You may not use this file
  * except in compliance with the Camunda License 1.0.
  */
-package io.camunda.zeebe.gateway.rest;
+package io.camunda.zeebe.gateway.rest.controller;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
 
 import io.camunda.zeebe.broker.client.api.BrokerClient;
 import io.camunda.zeebe.broker.client.api.BrokerClusterState;
 import io.camunda.zeebe.broker.client.api.BrokerTopologyManager;
-import io.camunda.zeebe.gateway.protocol.rest.BrokerInfo;
-import io.camunda.zeebe.gateway.protocol.rest.Partition;
-import io.camunda.zeebe.gateway.protocol.rest.Partition.HealthEnum;
-import io.camunda.zeebe.gateway.protocol.rest.Partition.RoleEnum;
-import io.camunda.zeebe.gateway.protocol.rest.TopologyResponse;
-import io.camunda.zeebe.gateway.rest.TopologyControllerTest.TestTopologyApplication;
 import io.camunda.zeebe.protocol.record.PartitionHealthStatus;
 import io.camunda.zeebe.util.VersionUtil;
 import java.util.List;
 import java.util.Set;
 import java.util.concurrent.ThreadLocalRandom;
 import org.junit.jupiter.api.BeforeEach;
-import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
 import org.mockito.Mockito;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.boot.autoconfigure.SpringBootApplication;
-import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.test.context.SpringBootTest.WebEnvironment;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
 import org.springframework.boot.test.mock.mockito.MockBean;
 import org.springframework.http.MediaType;
-import org.springframework.test.web.reactive.server.WebTestClient;
-
-@SpringBootTest(
-    classes = {TestTopologyApplication.class, TopologyController.class},
-    webEnvironment = WebEnvironment.RANDOM_PORT)
-public class TopologyControllerTest {
 
-  private static final String TOPOLOGY_BASE_URL = "v1/topology";
+@WebMvcTest(TopologyController.class)
+public class TopologyControllerTest extends RestControllerTest {
 
   @MockBean BrokerClient brokerClient;
   @MockBean BrokerTopologyManager topologyManager;
 
-  @Autowired private WebTestClient webClient;
-
   @BeforeEach
   void setUp() {
     Mockito.when(brokerClient.getTopologyManager()).thenReturn(topologyManager);
   }
 
-  @Test
-  public void shouldGetTopology() {
+  @ParameterizedTest
+  @ValueSource(strings = {"/v1/topology", "/v2/topology"})
```

Comment:
:+1: Cool! Preparing for the unified API I see :smiley:

Reply:
Yeah, exactly! 👍